src/Nix/Expr/Types.hs

```diff
@@ -226,10 +236,20 @@ data NString r
   = DoubleQuoted ![Antiquoted Text r]
   -- ^ Strings wrapped with double-quotes (") can contain literal newline
   -- characters, but the newlines are preserved and no indentation is stripped.
+  --
+  -- > DoubleQuoted [Plain "x",Antiquoted y]   ~  "x${y}"
   | Indented !Int ![Antiquoted Text r]
   -- ^ Strings wrapped with two single quotes ('') can contain newlines, and
   --   their indentation will be stripped, but the amount stripped is
   --   remembered.
+  --
+  -- > Indented 1 [Plain "x"]                  ~  '' x''
```

Comment:
I suspect this might be an unusual form to format indented strings. Would

```
''
 x''
```

be better?

Reply:
Slightly related to #531 - I think we should opt for canonical output like
```
''
  x
''
```

which should hopefuly be part of my fix for that issue. I guess keep it for now and we can adjust later.

I'm not entirely sure if it's a good idea but it makes sense to me, for one-liners utilizing two single quotes to avoid escaping we could have a special case where `Indented 0` would result in one-liner.

Reply:
I should clarify that I didn't attempt to show to the Nix expressions in the way `hnix` would prettyprint them here. Maybe I should though?! I was mostly aiming for a representation that is easy to understand.



> I think we should opt for canonical output like
> 
> ```
> ''
>   x
> ''
> ```

This string is actually not equivalent – it has an extra newline:

```
nix-repl> ''
           x
          ''
"x\n"
```

> I'm not entirely sure if it's a good idea but it makes sense to me, for one-liners utilizing two single quotes to avoid escaping we could have a special case where `Indented 0` would result in one-liner.

Sounds fine to me. I don't really have an opinion on this yet! ;)